app/lib/actions/katello/upstream_subscriptions/remove_entitlements.rb

```diff
@@ -3,17 +3,30 @@ module Katello
     module UpstreamSubscriptions
       class RemoveEntitlements < Actions::Base
         def plan(pool_ids = [])
+          organization = Object::Organization.current
+          url = "#{organization.owner_details['upstreamConsumer']['apiUrl']}#{organization.owner_details['upstreamConsumer']['uuid']}"
```

Comment:
the URL and Cert code here - I feel this knowledge should be limited to the resource class. Could this be moved to
`Katello::Resources::Candlepin::UpstreamConsumer.get` ?

Reply:
I think that's a good way to make this code reusable.  `UpstreamConsumer.get` already exists since it's inherited, did you mean I should overload that?

Reply:
ah! does that method already work then? doesn't look like we're using it anywhere so it might be broken in some way

Reply:
I haven't tried using `UpstreamConsumer.get` since I haven't seen us using it at all, I've only seem `UpstreamConsumer.resource(...).get` like I'm doing on line 13.
Maybe instead I could write some methods in `upstream_consumer.rb` to fetch the url, cert, and key?

Reply:
as far as I know most of the resource classes have a .get method so it may come to implementing one (for consistency & convenience)  to do exactly what you did here in the action